### PR TITLE
Prescription list sorting for desktop and mobile

### LINKED ISSUE #4411: Prescription list sorting for desktop and mobile
After the prescriptions design/ux audit with the core team, we discussed the same issue you guys brought up in this [PR about rx list view](https://github.com/department-of-veterans-affairs/vets-website/pull/4391) of how the sort option is not consistent. Mobile is missing ascending/descending option. 

Changes: 
`Last submit date` --> `Submit date   `
`Last fill date` --> `Fill day`
Next to each dropdown, add links to view Newest to Oldest and Oldest to Newest
screen shot 2016-12-15 at 8 19 10 pm

**Prescription refill page card view (desktop)**: Dropdown 
We are changing to dropdown because at most recent testing, users missed sort by links. 
- Prescription name (default)
- Facility name
- Submit date

screen shot 2016-12-15 at 8 17 49 pm

**Prescription refill page list view (desktop)**: Embedded sort in table
- Prescription name (default)
- Facility name
- Submit date
screen shot 2016-12-15 at 8 22 07 pm

**Prescription history page (desktop)**: Embedded sort in table
screen shot 2016-12-15 at 8 23 20 pm
- Submit date (default)
- Prescription name
- Fill date
- Prescription status

**Prescription refill page (mobile)**: Dropdown 
screen shot 2016-12-15 at 8 26 59 pm
- Prescription name (default)
- Facility name
- Submit date

**Prescription history page (mobile)**: Dropdown 
screen shot 2016-12-15 at 8 27 11 pm
- Submit date (default)
- Prescription name
- Fill date
- Prescription status

I wanted to break it down by pages and device. Before breaking this down to smaller issues, wanted to give @U-DON and @bshyong a bigger picture context since it feels like a lot of the issues are intertwined.

### PR DESCRIPTION
**Pull request by bshyong**
- resolves https://github.com/department-of-veterans-affairs/vets-website/issues/4411

### CODE REVIEW

File: src/js/rx/components/SortMenu.jsx [hunk 1]
```diff
@@ -21,35 +21,41 @@ class SortMenu extends React.Component {
...
-        return (
-          <li key={ind}>
-            <a
-                className={selectedClass}
-                onClick={this.handleClick(o.value)}>
-              {o.label}
-            </a>
-          </li>
-        );
+    const options = this.props.selected.value === 'lastSubmitDate' ? {
```
Comment (U-DON) [inline src/js/rx/components/SortMenu.jsx]: You've already pulled out `value` in the destructuring above, so could just do `selected.value === ...`.

**Review by U-DON — COMMENTED** [review]
LGTM. This adds the new sort for Active, but I think History also needs the change.

**Comment by bshyong** [comment]
I was wondering what you meant by History needing the change—I'll add that. The desktop view for History is fine but the mobile view needs the directional sort links as well...

**Comment by bshyong** [comment]
Latest changes added

File: src/js/rx/components/SortMenu.jsx [hunk 2]
```diff
@@ -21,35 +21,45 @@ class SortMenu extends React.Component {
...
-                className={selectedClass}
-                onClick={this.handleClick(o.value)}>
-              {o.label}
-            </a>
-          </li>
-        );
+    const options = ['refillSubmitDate',
+      'lastSubmitDate',
+      'refillDate',
+      'refillStatus'
```
Comment (U-DON) [inline src/js/rx/components/SortMenu.jsx]: Should `refillStatus` be associated with 'Newest to Oldest'/'Oldest to Newest'? Seems 'A-Z'/'Z-A' would be more appropriate, although there is that tricky issue of the display names not falling into exact alphabetical order.

File: src/js/rx/containers/Active.jsx [hunk 3]
```diff
@@ -118,11 +118,11 @@ class Active extends React.Component {
                 onChange={this.handleSort}
                 onClick={this.handleSort}
                 options={sortOptions}
-                selected={sortValue.value}/>
+                selected={sortValue}/>
```
Comment (U-DON) [inline src/js/rx/containers/Active.jsx]: Why are `sortValue` and `currentSort` above both assigned with `this.props.sort`? I'm guessing it should just be `currentSort`.

File: src/js/rx/containers/History.jsx [hunk 4]
```diff
@@ -146,10 +146,13 @@ class History extends React.Component {
       content = (
         <div>
           <p className="rx-tab-explainer">Your VA prescription refill history.</p>
-          <SortMenu
-              onChange={this.handleSort}
-              options={fields}
-              selected={currentSort.value}/>
+          <div className="show-for-small-only">
```
Comment (U-DON) [inline src/js/rx/containers/History.jsx]: We've generally been using `@media (max-width: $medium-screen)` within the stylesheets to style the mobile-only content. Should we be using `show-for-small-only` classes instead?

**Review by U-DON — COMMENTED** [review]
Generally still LGTM with some questions.

File: src/js/rx/containers/History.jsx (hunk 4 above)
Comment (U-DON) [inline src/js/rx/containers/History.jsx]: Okay, works for me. I didn't know about it before, but it can be useful for some one-off cases like this.

**Review by U-DON — APPROVED** [review]

**Pull request merged**